File: AnimalHealthBookApi/AnimalHealthBookApi/Models/Appointment.cs
using Microsoft.EntityFrameworkCore;

namespace AnimalHealthBookApi.Models
{
    public class Appointment
    {
        public Guid Id { get; set; }

        public Guid VetId { get; set; }

        public Guid ClinicId { get; set; }
        public Guid AnimalId { get; set; }

        public DateTime Date { get; set; }

        public string? Description { get; set; }

        public virtual IEnumerable<Procedure> Procedures { get; set; }
        [DeleteBehavior(DeleteBehavior.NoAction)]

        public virtual Vet Vet { get; set; }

        [DeleteBehavior(DeleteBehavior.NoAction)]
        public virtual Clinic Clinic { get; set; }
        [DeleteBehavior(DeleteBehavior.NoAction)]

        public virtual Animal Animal { get; set; }
    }
}

File: animal-health-book/src/app/shared/models/appointment.ts
import { Procedure } from "./procedure";
import { Vet } from "./vet";
import { Clinic } from "./clinic";
import { Animal } from "./animal";

export interface Appointment {
    id: string;
    vetId: string;
    clinicId: string;
    animalId: string;
    date: string;
    description: string | null;
    procedures: Procedure[];
    vet: Vet;
    clinic: Clinic;
    animal: Animal;
}